src/backend/distributed/planner/multi_join_order.c

```diff
@@ -1060,45 +1068,50 @@ SinglePartitionJoin(JoinOrderNode *currentJoinNode, TableEntry *candidateTable,
  * clause exists, the function returns NULL.
  */
 OpExpr *
-SinglePartitionJoinClause(Var *partitionColumn, List *applicableJoinClauses)
+SinglePartitionJoinClause(List *partitionColumns, List *applicableJoinClauses)
 {
-	if (partitionColumn == NULL)
+	if (list_length(partitionColumns) == 0)
 	{
 		return NULL;
 	}
 
-	Node *applicableJoinClause = NULL;
-	foreach_ptr(applicableJoinClause, applicableJoinClauses)
+	Var *partitionColumn = NULL;
+	foreach_ptr(partitionColumn, partitionColumns)
 	{
-		if (!NodeIsEqualsOpExpr(applicableJoinClause))
-		{
-			continue;
-		}
-		OpExpr *applicableJoinOpExpr = castNode(OpExpr, applicableJoinClause);
-		Var *leftColumn = LeftColumnOrNULL(applicableJoinOpExpr);
-		Var *rightColumn = RightColumnOrNULL(applicableJoinOpExpr);
-		if (leftColumn == NULL || rightColumn == NULL)
+		Node *applicableJoinClause = NULL;
```

Comment:
Minor: Moving the inner block to a function could help readability. Similar for `JoinOnColumns ()`.